packages/ERTP/src/paymentLedger.js

```diff
@@ -156,18 +158,18 @@ export const preparePaymentLedger = (
    * - Every payment that is a key in the outer `paymentRecoverySets` weakMap is
    *   also in the recovery set indexed by that payment.
    * - Implied by the above but worth stating: the payment is only in at most one
-   *   recovery set.
+   *   one recovery set.
    * - A recovery set only contains such payments.
    * - Every purse is associated with exactly one recovery set unique to it.
    * - A purse's recovery set only contains payments withdrawn from that purse and
    *   not yet consumed.
    *
-   * @type {WeakMapStore<Payment, SetStore<Payment>>}
+   * @type {WeakMapStore<Payment, SetStore<Payment>> | undefined}
    */
-  const paymentRecoverySets = provideDurableWeakMapStore(
-    issuerBaggage,
-    'paymentRecoverySets',
-  );
+  const paymentRecoverySets =
+    recoverySetOption === 'noRecoverySets'
+      ? undefined
+      : provideDurableWeakMapStore(issuerBaggage, 'paymentRecoverySets');
 
```

Comment:
What if the revived issuer has `'noRecoverySets'` but the old state has recovery sets?